[HOLD] Universal metrics export preview (#3880)
astronomer/docs: astro/universal-metrics-export.md

---
sidebar_label: "Export metrics"
title: "Export metrics from Astro"
id: export-metrics
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import TeamBadge from '@site/src/components/TeamBadge';
import HostedBadge from '@site/src/components/HostedBadge';

<TeamBadge/>

:::publicpreview
:::

> You can export comprehensive metrics about your Astro Deployments from Astro directly to any third-party monitoring and alerting systems. The universal metrics exporter uses the [Prometheus data model](https://prometheus.io/docs/concepts/data_model/) format using the remote-write capability to ensure that measurements about your Astro Deployments to your preferred monitoring tools.

Reviewer: ```suggestion
You can export comprehensive metrics about your Astro Deployments from Astro directly to any third-party monitoring and alerting systems. The universal metrics exporter uses the [Prometheus data model](https://prometheus.io/docs/concepts/data_model/) format using the remote-write capability to export metrics about your Astro Deployments to your preferred monitoring tools.
```

## Metric types

> - Airflow application level metrics: Metrics defined by Airflow, or custom metrics you write, about the health, success, and performance of data pipelines orchestrated and executed by Airflow. You can see the specific list of [application metrics](https://github.com/astronomer/ap-vendor/blob/main/statsd-exporter/include/mappings-gen2.yml) and read their corresponding descriptions in the [Airflow metrics descriptions](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/metrics.html#metric-descriptions).

Reviewer: ```suggestion
- Airflow application level metrics: Metrics defined by Airflow, about the health, success, and performance of data pipelines orchestrated and executed by Airflow. You can see the specific list of [application metrics](https://github.com/astronomer/ap-vendor/blob/main/statsd-exporter/include/mappings-gen2.yml) and read their corresponding descriptions in the [Airflow metrics descriptions](https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/logging-monitoring/metrics.html#metric-descriptions).
```

Reviewer: don't have custom metric support yet

> - Infrastructure level metrics: Metrics about running pods for the different Airflow components. These indicate of the use, health, and performance of the pods based on your workload and pipelines.

Reviewer: ```suggestion
- Infrastructure level metrics: Metrics about running pods for the different Airflow components. These indicate the usage, health, and performance of the pods based on your workload and pipelines.
```

  | Name                                        | Description           |
  | ------------------------------------------- | --------------------- |

>   | `container_cpu_usage_seconds_total`         | CPU use               |

Reviewer: ```suggestion
  | `container_cpu_usage_seconds_total`         | CPU usage               |
```

>   | `container_memory_working_set_bytes`        | Memory use            |

Reviewer: ```suggestion
  | `container_memory_working_set_bytes`        | Memory usage            |
```

>   | `kubelet_stats_ephemeral_storage_pod_usage` | Ephemeral storage use |

Reviewer: ```suggestion
  | `kubelet_stats_ephemeral_storage_pod_usage` | Ephemeral storage usage |
```

  | `kube_pod_status_*`                         | Kubernetes pod status |
  | `kube_pod_labels`                           | Kubernetes pod label  |

- Metadata labels: Both application and infrastructure metrics have the following metadata labels associated with them.
  - `cloud_provider`
  - `cloud_region`
  - `cluster_organization_id`
  - `container`
  - `namespace`
  - `pod`
  - `deploymentId`
  - `organizationId`
  - `workspaceId`

## Prerequisites

> - Bearer token or license key of your target data observability server

Reviewer: ```suggestion
- Supported Auth: Bearer token or license key, username and password, or custom HTTP header(s) of your target data observability server _(Optional)_
```

---
sidebar_label: 'Universal Metrics Export'
title: 'Export metrics from Astro'
id: metrics-export
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import TeamBadge from '@site/src/components/TeamBadge';
import HostedBadge from '@site/src/components/HostedBadge';

<TeamBadge/>

:::publicpreview
:::

> You can export comprehensive metrics about your Astro Deployments from Astro directly to any third-party monitoring and alerting systems. The universal metrics exporter uses the[OpenTelemetry (OTel)](https://opentelemetry.io/docs/) protocol to ensure that measurements about your Astro Deployments can be exported and used across any OTel compatible tool.

Author: ```suggestion
You can export comprehensive metrics about your Astro Deployments from Astro directly to any third-party monitoring and alerting systems. The universal metrics exporter uses the [Prometheus]() format using the remote-write capability to ensure that measurements about your Astro Deployments to your preferred monitoring toolsl.
```

> 
> This functionality allows you to move operational data about your Astro Deployments to your prefferred monitoring tools, like Prometheus.

Author: ```suggestion
```
